internal/tools/dataplex/dataplexsearchaspecttypes/dataplexsearchaspecttypes.go

```diff
@@ -0,0 +1,186 @@
+	Description  string   `yaml:"description"`
+	AuthRequired []string `yaml:"authRequired"`
+}
+
+// validate interface
+var _ tools.ToolConfig = Config{}
+
+func (cfg Config) ToolConfigKind() string {
+	return kind
+}
+
+func (cfg Config) Initialize(srcs map[string]sources.Source) (tools.Tool, error) {
+	// Initialize the search configuration with the provided sources
+	rawS, ok := srcs[cfg.Source]
+	if !ok {
+		return nil, fmt.Errorf("no source named %q configured", cfg.Source)
+	}
+	// verify the source is compatible
+	s, ok := rawS.(compatibleSource)
+	if !ok {
+		return nil, fmt.Errorf("invalid source for %q tool: source kind must be one of %q", kind, compatibleSources)
+	}
+
+	query := tools.NewStringParameter("query", "The query against which aspect type should be matched.")
+	pageSize := tools.NewIntParameterWithDefault("pageSize", 5, "Number of results in the search page.")
+	pageToken := tools.NewStringParameterWithDefault("pageToken", "", "Page token received from a previous search aspect types tool call. Provide this to retrieve the subsequent page.")
+	orderBy := tools.NewStringParameterWithDefault("orderBy", "relevance", "Specifies the ordering of results. Supported values are: relevance, last_modified_timestamp, last_modified_timestamp asc")
+	semanticSearch := tools.NewBooleanParameterWithDefault("semanticSearch", true, "Whether to use semantic search for the query. If true, the query will be processed using semantic search capabilities.")
+	parameters := tools.Parameters{query, pageSize, pageToken, orderBy, semanticSearch}
+
+	mcpManifest := tools.McpManifest{
+		Name:        cfg.Name,
+		Description: cfg.Description,
+		InputSchema: parameters.McpManifest(),
+	}
+
+	t := &Tool{
+		Name:          cfg.Name,
+		Kind:          kind,
+		Parameters:    parameters,
+		AuthRequired:  cfg.AuthRequired,
+		CatalogClient: s.CatalogClient(),
+		ProjectID:     s.ProjectID(),
+		manifest: tools.Manifest{
+			Description:  cfg.Description,
+			Parameters:   parameters.Manifest(),
+			AuthRequired: cfg.AuthRequired,
+		},
+		mcpManifest: mcpManifest,
+	}
+	return t, nil
+}
+
+type Tool struct {
+	Name          string
+	Kind          string
+	Parameters    tools.Parameters
+	AuthRequired  []string
+	CatalogClient *dataplexapi.CatalogClient
+	ProjectID     string
+	manifest      tools.Manifest
+	mcpManifest   tools.McpManifest
+}
+
+func (t *Tool) Authorized(verifiedAuthServices []string) bool {
+	return tools.IsAuthorized(t.AuthRequired, verifiedAuthServices)
+}
+
+var getAspectMap = map[string]*dataplexpb.AspectType{}
+
+func (t *Tool) Invoke(ctx context.Context, params tools.ParamValues) (any, error) {
+	paramsMap := params.AsMap()
+	query, _ := paramsMap["query"].(string)
+	pageSize := int32(paramsMap["pageSize"].(int))
+	pageToken, _ := paramsMap["pageToken"].(string)
+	orderBy, _ := paramsMap["orderBy"].(string)
+	semanticSearch, _ := paramsMap["semanticSearch"].(bool)
+
+	req := &dataplexpb.SearchEntriesRequest{
```

Comment:
Please add comments so those unfamiliar with dataplex can help maintain this code.